bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/profiles/SystemOffsetProfile.java

```diff
@@ -0,0 +1,137 @@
+import org.eclipse.smarthome.core.thing.profiles.StateProfile;
+import org.eclipse.smarthome.core.thing.profiles.SystemProfiles;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tec.uom.se.AbstractUnit;
+
+/**
+ * Applies the given parameter "offset" to a QuantityType or DecimalType state
+ *
+ * @author Stefan Triller - initial contribution
+ *
+ */
+@NonNullByDefault
+public class SystemOffsetProfile implements StateProfile {
+
+    private static Logger logger = LoggerFactory.getLogger(SystemOffsetProfile.class);
+    private static final String OFFSET_PARAM = "offset";
+
+    private final ProfileCallback callback;
+    private final ProfileContext context;
+
+    @Nullable
+    private QuantityType<?> offset;
+
+    public SystemOffsetProfile(ProfileCallback callback, ProfileContext context) {
+        this.callback = callback;
+        this.context = context;
+
+        Object paramValue = this.context.getConfiguration().get(OFFSET_PARAM);
+        logger.debug("Parameter '{}' has value '{}' ", OFFSET_PARAM, paramValue);
+        if (paramValue instanceof String) {
+            try {
+                offset = new QuantityType<>((String) paramValue);
+            } catch (IllegalArgumentException e) {
+                logger.error("Cannot convert value '{}' of parameter '{}' into a valid offset of type QyantityType.",
+                        paramValue, OFFSET_PARAM);
+            }
+        } else {
+            logger.error("Parameter '{}' is not of type String", OFFSET_PARAM);
+        }
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return SystemProfiles.OFFSET;
+    }
+
+    @Override
+    public void onStateUpdateFromItem(State state) {
+        logger.debug("State update from item: '{}'", state);
+        callback.handleCommand((Command) applyOffset(state, false));
+    }
+
+    @Override
+    public void onCommandFromItem(Command command) {
+        logger.debug("Command from item: '{}'", command);
+        callback.handleCommand((Command) applyOffset(command, false));
+    }
+
+    @Override
+    public void onCommandFromHandler(Command command) {
+        logger.debug("Command from handler: '{}'", command);
+        callback.sendCommand((Command) applyOffset(command, true));
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State state) {
+        logger.debug("State update from handler: '{}'", state);
+        callback.sendUpdate((State) applyOffset(state, true));
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    private Type applyOffset(Type state, boolean towardsThing) {
+        QuantityType finalOffset = offset;
+        if (finalOffset == null) {
+            logger.error("Offset not configured correctly, returning undef state.");
```

Comment:
This just states what is is doing - as logs are directed towards (admin) users, do you think this could be made a little more helpful wrt to guiding the user to do it right?